.github/workflows/test-plugin.yml

```diff
@@ -83,3 +83,76 @@ jobs:
     - name: Run CPU backend examples
       working-directory: triton_shared/python/examples
       run: pytest .
+
+
+  build_and_test_triton_shared_arm:
+    runs-on: ["self-hosted", "1ES.Pool=triton-shared-github-arm"]
+
+    steps:
+    - name: Check Architecture
+      run: uname -a
+
+    - name: Force Failure
+      if: ${{ inputs.force-failure }}
+      run: exit 1
+
+
+    - name: Checkout Triton-Shared
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.triton-shared-ref }}
+        path: triton_shared
+        submodules: recursive
+
+    - name: Clear Triton Cache
+      run: |
+        rm -rf ~/.triton
+        
+    - name: Modify Instance
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y build-essential ninja-build python3-pip libc6
+    - name: Update PATH
+      run: |
+        echo "PATH=${HOME}/.local/bin:${PATH}" >> "${GITHUB_ENV}"
+    - name: Check pre-commit
+      working-directory: triton_shared/triton
+      run: |
+        python3 -m pip install --upgrade pre-commit
+        python3 -m pre_commit run --all-files --verbose
+    - name: Edit setup for arm64
+      working-directory: triton_shared/triton
+      run: |
+        sed -i 's/arch = {"x86_64": "64", "arm64": "aarch64"}/arch = {"x86_64": "64", "arm64": "aarch64", "aarch64":"aarch64"}/' "python/setup.py"
+        sed -i 's|https://tritonlang.blob.core.windows.net/llvm-builds/|https://storage.googleapis.com/compiled-blob/|g' "python/setup.py"
```

Comment:
Is this needed after your PR to xcompile the LLVM binaries?

Reply:
I think so until the upstream updates the LLVM version. Although I updated it, don't think it was ram

Reply:
The arm64 ubuntu build is running and creating the tgz file.

https://github.com/openai/triton/actions/runs/7960036148/job/21728181209

/home/runner/work/triton/triton/llvm-3e77871c-ubuntu-arm64.tar.gz

The job name says Ubuntu 20.04 and could be updated to say Ubuntu 20.04 Arm64.

Reply:
looks like the upstream cross compiler isnt working need to fix

https://github.com/openai/triton/pull/3134#issuecomment-1957239635
unsure why since

Reply:
Getting close! I was able to build Triton/Triton-shared with the new tgz on Ubuntu Arm64.

Reply:
Looks like OAI updated the llvm binaries for arm64 and you can undo the url change

Reply:
looks like the entire llvm build was cancelled from some odd macos x64 failure. Don't think my PR would have caused that

Reply:
I believe the bucket has been updated and should be able to be re-ran